### PR TITLE
add reduce with ObjectIterator

### PR DESCRIPTION
**Pull request by davetayls**
I've added underscore reduce definition with the ObjectIterator

### CODE REVIEW

**Comment by vvakame** [comment]
@davetayls thank you your contributions.
sorry, I can't found your change in the document.
do you have any pointer?

**Comment by davetayls** [comment]
hey it's in underscore/underscore.d.ts line 182... i've added another reduce definition

**Comment by vvakame** [comment]
@davetayls oh, sorry. My english is too bad.
I can't review your PR.
your usecase can't find from http://underscorejs.org/#reduce 
Do you know the resources available to help in the review?

**Comment by davetayls** [comment]
You will see in that doc that it says:

> The iteratee is passed four arguments: the memo, then the value and index **(or key)** of the iteration, and finally a reference to the entire list.

The important part is that the list can be an array or an object. If the *iteratee* is an object then it passes the key. The `ObjectIterator` was already defined but wasn't used here so I just added it in

**Comment by vvakame** [comment]
oh, I see.

File: underscore/underscore.d.ts [hunk 1]
```diff
@@ -179,6 +179,12 @@ interface UnderscoreStatic {
 		memo?: TResult,
 		context?: any): TResult;
 
+	reduce<T, TResult>(
+		list: _.Collection<T>,
```
Comment (vvakame) [inline underscore/underscore.d.ts]: I think `object: _.Dictionary<T>` is rihgt, not `list: _.Collection<T>`.
Is this right?

Comment (davetayls) [inline underscore/underscore.d.ts]: ah yes, i'll update that

**Comment by davetayls** [comment]
I've rebased on master and updated to use a Dictionary on the list property

**Comment by vvakame** [comment]
@davetayls thanks mate!

**Comment by davetayls** [comment]
no worries :)

**Pull request merged**